File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/radius/RadiusRoles.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.radius

import androidx.compose.runtime.Immutable
import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.atomic._primitives.sizing.SizingPrimitives

@Immutable
data class RadiusRoles(
    val xs: Dp,
    val s: Dp,
    val m: Dp,
    val l: Dp,
    val xl: Dp,
) {
    companion object {
        val Default = RadiusRoles(
            xs = SizingPrimitives.Size2,
            s = SizingPrimitives.Size4,
            m = SizingPrimitives.Size8,
            l = SizingPrimitives.Size12,
            xl = SizingPrimitives.Size20,
        )
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Radius/RadiusRoles.swift
import Foundation

public struct RadiusRoles {
    public let xs: CGFloat
    public let s: CGFloat
    public let m: CGFloat
    public let l: CGFloat
    public let xl: CGFloat

    public static let `default` = RadiusRoles(
        xs: SizingPrimitives.size2.rawValue,
        s: SizingPrimitives.size4.rawValue,
        m: SizingPrimitives.size8.rawValue,
        l: SizingPrimitives.size12.rawValue,
        xl: SizingPrimitives.size20.rawValue
    )
}
